### ISSUE TITLE
TTSLua

### ISSUE THREAD

**Issue by Kijan** [Koihik/vscode-lua-format#16]
Hi 
I am new to VSC and wanted to use your formatter for files with the extension *.ttslua.
But i cant get it to work. Can you tell me what i have to change to get to work?

**Comment by karanankit01** [comment]
- [ ] First of all install the extension
- [ ] Then right click on the .lua file
- [ ] After that select "Format document with..."
- [ ] At last select "vscode-lua-format"

**Comment by Kijan** [comment]
no i think you missunderstood me.
i have it installed in VSC and when i have lua files and press ALT+SHIFT+F it works like it should ;)
but i have *.ttslua and there i get every time 
"There is no formatter for 'TTS Lua' files installed."
so thats why i thought your formatter is not set to ttslua and maybe that could be changed easily like i did it for Atom editor.
but i cant get it to work for VSC -.-

**Comment by Koihik** [comment]
```
"files.associations": {
    "*.ttslua": "lua"
}
```
You can add this in your settings.json. VSC will treat `.ttslua` as `.lua`.

**Comment by Kijan** [comment]
That is not a good alternative because i am using an other extension for the ttslua with autofill and other stuff that will not work when it is triggert as lua sorry :(

**Comment by Koihik** [comment]
> That is not a good alternative because i am using an other extension for the ttslua with autofill and other stuff that will not work when it is triggert as lua sorry :(

```
"[ttslua]": {
    "editor.defaultFormatter": "Koihik.vscode-lua-format"
},
```
specify LuaFormatter as default formatter for any extension you want.
btw: LuaFormatter only works when the file is a valid Lua source file.

**Comment by Kijan** [comment]
i cant get it to work :(
`    "files.associations": {
        "*.ttslua": "TTS Lua"
    },
    "[ttslua]": {
        "editor.defaultFormatter": "Koihik.vscode-lua-format"
    },
    "[TTS Lua]": {
        "editor.defaultFormatter": "Koihik.vscode-lua-format"
    },
    "[TTSLua]": {
        "editor.defaultFormatter": "Koihik.vscode-lua-format"
    },`
i tested different things...
when i change "*.ttslua": "TTS Lua" to lua then your formatter works but the other extension not any more...

**Comment by Koihik** [comment]
> i cant get it to work :(
> ` "files.associations": { "*.ttslua": "TTS Lua" }, "[ttslua]": { "editor.defaultFormatter": "Koihik.vscode-lua-format" }, "[TTS Lua]": { "editor.defaultFormatter": "Koihik.vscode-lua-format" }, "[TTSLua]": { "editor.defaultFormatter": "Koihik.vscode-lua-format" },`
> i tested different things...
> when i change "*.ttslua": "TTS Lua" to lua then your formatter works but the other extension not any more...

Each language has a unique identifier in VSCode. [Language Identifiers](https://code.visualstudio.com/docs/languages/identifiers)
If other extension works well, you can check language identifier on status bar.

**Comment by Kijan** [comment]
yeah if i click at the bottom then at the top there is "(TTS Lua)" next to "Tabletop Simulator Lua"... 
thats why i tested this but it is not working...
I stopped to try getting it to work.
